consensus/poet/core/sawtooth_poet/poet_consensus/poet_block_verifier.py

```diff
@@ -184,6 +184,65 @@ def verify_block(self, block_wrapper):
                             validator_state.key_block_claim_count,
                             poet_config_view.key_block_claim_limit))
 
+            # While having a block claim delay is nice, it turns out that in
+            # practice the claim delay should not be more than one less than
+            # the number of validators.  It helps to imagine the scenario
+            # where each validator hits their block claim limit in sequential
+            # blocks and their new validator registry information is updated
+            # in the following block by another validator, assuming that there
+            # were no forks.  If there are N validators, once all N validators
+            # have updated their validator registry information, there will
+            # have been N-1 block commits and the Nth validator will only be
+            # able to get its updated validator registry information updated
+            # if the first validator that kicked this off is now able to claim
+            # a block.  If the block claim delay was greater than or equal to
+            # the number of validators, at this point no validators would be
+            # able to claim a block.
+            number_of_validators = \
+                len(validator_registry_view.get_validators())
+            block_claim_delay = \
+                min(
+                    poet_config_view.block_claim_delay,
+                    number_of_validators - 1)
+
+            # While a validator network is starting up, we need to be careful
+            # about applying the block claim delay because if we are too
+            # aggressive we will get ourselves into a situation where the
+            # block claim delay will prevent any validators from claiming
+            # blocks.  So, until we get at least block_claim_delay blocks
+            # we are going to choose not to enforce the delay.
+            if consensus_state.total_block_claim_count <= block_claim_delay:
+                LOGGER.debug(
+                    'Skipping block claim delay check.  Only %d block(s) in '
+                    'the chain.  Claim delay is %d block(s). %d validator(s) '
+                    'registered.',
+                    consensus_state.total_block_claim_count,
+                    block_claim_delay,
+                    number_of_validators)
+                return True
+
+            blocks_since_registration = \
+                block_wrapper.block_num - \
+                validator_state.commit_block_number - 1
+
+            if block_claim_delay > blocks_since_registration:
+                raise \
```

Comment:
To be consistent with the previous PoET policy test that also raises ValueError on rejecting block.